cpp/include/cudf/table/table.hpp

```diff
@@ -112,6 +112,28 @@ class table {
    *---------------------------------------------------------------------------**/
   table_view select(std::vector<cudf::size_type> const& column_indices) const;
 
+  /**---------------------------------------------------------------------------*
+   * @brief Returns a reference to the specified column
+   *
+   * @throws std::out_of_range
+   * If i is out of the range [0, num_columns)
+   *
+   * @param i Index of the desired column
+   * @return A reference to the desired column
+   *---------------------------------------------------------------------------**/
+  column& get_column(cudf::size_type column_index) { return *(_columns.at(column_index)); }
```

Comment:
It seems weird to me that this method is called `column` in table_view/table_device_view, but it's called `get_column` here. Maybe we should make these consistent?

Reply:
Good catch. If I call it `column`, I run into https://stackoverflow.com/questions/15537023/declaration-of-method-changes-meaning-of-symbol due to there already being a class `column`.

So we'd have to change to `get_column` in `table_(device)_view`. Is that something that should be tackled in this PR?

Reply:
@jrhemstad what do you think about `get_column()` vs `column()`?
It would probably be better to make that change in a separate PR since `column()` is used in quite a few places?

Reply:
Yeah I felt it was orthogonal to this PR. I'm happy to submit a follow-up

Reply:
Rename it all to `get_column` since `column()` won't work in `table`.